db/utils.js

```diff
@@ -134,6 +152,16 @@ const DBUtils = {
       return await aEntry
         .$relatedQuery("breaches")
         .orderBy("name");
+    }, async () => {
+      return [];
+    });
+  },
+
+  async getBreachesForHashPrefix(sha1Prefix) {
+    return await this._getSha1EntriesFromPrefixAndDo(sha1Prefix, async aEntries => {
```

Comment:
Is this the only time we do this? The naming of _getSha1EntriesFromPrefixAndDo is confusing to me based on what it does. It seems like what it's really doing is getting *breaches*? and the two callbacks do the same thing - we can just:

a) get rid of this function,
b) rename _getSha1EntriesFromPrefixAndDo to getBreachesForHashPrefix,
c) make it just return either the entries or the empty array

and things would still work the same, no? Let me know if I'm missing something :)

Reply:
Yeah these functions confused me. (I based this function on the existing `_getSha1EntryAndDo` function) I started to re-write it but quickly gave up since this prefix-only version of the function is only used by the hibp stub API, which we should be able to throw away.